### DISCUSSION TITLE
Whisper should support mixed languages

### DISCUSSION THREAD

**Discussion by toanhn** [openai/whisper#2385] [Ideas]
I have an audio file, a speaker speaks two languages Chinese and English.
When I use Microsoft Cognitive, I see it supports mixed languages, I call its APIs with two candidate languages Chinese and English, and Cognitive can capture Chinese and English sentences in the audio file well.
I think it would be great if Whisper could support mixed languages like that.
Thank you so much to the Whisper team and all of you.

**Comment by edsu** [comment]
I'm curious, have you tried not specifying the language on this audio and seeing how well Whisper performs?

**Reply by toanhn** [reply]
Yes, it will take the first 30 seconds of the audio file to detect the spoken language. But it is sometimes incorrect since besides detecting the content is spoken it also detects the speaker's accent. Ex: A Chinese speaks English can be detected as Chinese instead of English.
So the auto language detection is not reliable.
You can refer this https://github.com/openai/whisper/discussions/1644
